src/Documentation/resources/Best_Practices.md

```diff
@@ -0,0 +1,422 @@
+# Best Practices
+
+Orleans was designed in 2014 with the intention of removing low-level
+abstractions from actor model frameworks. Grains (actors stored in memory) are
+used to maintain the lifecycle among other features. Orleans is based on the
+actor model and fits some projects better that others.
+
+## Orleans should be considered when:
+
+-   Significant number (hundreds to millions) of loosely coupled entities
+
+    -   Example: Internet of Things (IoT) – individual devices turning on/off
+
+-   Entities are small enough to be single-threaded
+
+    -   Example: Determine if stock should be purchased based on current price
+
+-   Workload is interactive
+
+    -   Example: request-response, start/monitor/complete
+
+-   More than one server is expected or may be required\\
+
+    -   Orleans runs on a cluster which is expanded by adding servers to expand
+        the cluster
+
+-   Global coordination is not needed or on a smaller scale between a few
+    entities at a time
+
+    -   Orleans maintains its clusters and talks to other clusters. Each cluster
+        is able to communicate through tcp/ip
+
+-   *Various entities are used at different times*
+
+    -   This depends on the entities that are being used
+
+## Orleans is not the best fit when:
+
+-   Memory must be shared between entities
+
+    -   Each grain maintains its own states and should not be shared.
+
+-   A small number of large entities that may be multithreaded
+
+    -   A microservice may be a better option when supporting complex logic in a
+        single service
+
+-   Global coordination and/or consistency is needed
+
+    -   Orleans maintains the silos within its own framework
```

Comment:
Here again the example doesn't seem to fit the topic. Maybe "Such global coordination would severely limit performance of an Orleans-based application."?

Reply:
corrected